packages/frontend/src/components/common/GoogleLoginButton.tsx

```diff
@@ -0,0 +1,30 @@
+import { AnchorButton } from '@blueprintjs/core';
+import React from 'react';
+import { useFlashMessages } from '../../hooks/useFlashMessages';
+import { useApp } from '../../providers/AppProvider';
+
+export const GoogleLoginButton: React.FC = () => {
+    const { health, showToastError } = useApp();
+    const flashMessages = useFlashMessages();
+
+    if (!health.data?.auth.google.oauth2ClientId) {
+        return null;
+    }
+
+    if (flashMessages.data?.error) {
+        showToastError({
+            title: 'Failed to authenticate',
+            subtitle: flashMessages.data.error.join('\n'),
+        });
+    }
+
+    return (
+        <AnchorButton
+            href={`/api/v1${
+                health.data.auth.google.loginPath
+            }?redirect=${encodeURIComponent(window.location.href)}`}
```

Comment:
Yes good point!! 

Using `href` instead of `path` was only to support the dev environment.

But even if we can avoid `path` it would be great. Do you have any suggestions for passing the current front-end page to the backend session. This gets picked up later after the entire oauth flow is complete (involving a redirect away from the app, so FE unmounts completely, and 2 backend calls are made).

Reply:
> Using href instead of path was only to support the dev environment.
FE defines the `redirectPath` and the BE redirects to `siteUrl`+`redirectPath`